### ISSUE TITLE
Unified VanillaTransformer with compiled weights

### ISSUE THREAD

**Issue by oaustegard** [oaustegard/llm-as-computer#46]
## Summary

Build a single `VanillaTransformer(nn.Module)` matching Percepta's architecture — standard `nn.MultiheadAttention` layers, gated FF (`relu(gate) * val`), residual connections — where **all weight matrices are set analytically** (not trained). The model's `forward(idx)` takes a sequence of token IDs and returns next-token logits.

This replaces our per-head `CompiledAttentionHead` approach with standard PyTorch MHA, and replaces our Python if/elif opcode dispatch with a compiled FF layer.

## Architecture Target

```python
class VanillaTransformer(nn.Module):
    def __init__(self, vocab, d_model, n_heads, n_layers, d_ffn):
        self.tok = nn.Embedding(vocab, d_model)        # From Issue: Token Vocab
        self.attn = nn.ModuleList([nn.MultiheadAttention(...)])
        self.ff_in = nn.ModuleList([nn.Linear(d_model, 2*d_ffn)])
        self.ff_out = nn.ModuleList([nn.Linear(d_ffn, d_model)])
        self.head = nn.Linear(d_model, vocab)           # Unembedding
    def forward(self, idx):
        x = self.tok(idx) + pos_emb(T)
        for attn, ff_in, ff_out in zip(...):
            y, _ = attn(x, x, x, attn_mask=causal)
            x = x + y
            gate, val = ff_in(x).chunk(2, dim=-1)
            x = x + ff_out(F.relu(gate) * val)
        return self.head(x)
```

## Acceptance Criteria

- [ ] `VanillaTransformer` class with `forward(idx: LongTensor) → logits` matching Percepta's signature
- [ ] Uses `nn.MultiheadAttention` (not custom `CompiledAttentionHead` modules)
- [ ] Uses gated FF: `gate, val = ff_in(x).chunk(2); x += ff_out(relu(gate) * val)`
- [ ] All weights compiled analytically (no gradient descent)
- [ ] Our 10 active attention heads packed into standard MHA Q/K/V weight matrices
- [ ] Opcode dispatch + arithmetic compiled into FF weight matrices
- [ ] `argmax(model.forward(program_tokens))` produces correct next token for Phase 4 test programs (PUSH/POP/ADD/DUP/HALT) — at minimum 10/10 match
- [ ] Extended ISA coverage: SUB, JZ/JNZ, SWAP/OVER/ROT produce correct next tokens
- [ ] Parameter count reported and compared against current CompiledModel (currently 964 compiled params)

## Key Challenges

1. **Packing attention heads into MHA**: Our 10 custom heads (prog_op, prog_arg, stack_a/b/c, local_val/addr, heap_val/addr, call_stack) use different Q/K/V projections. Must pack them into standard `in_proj_weight` format where Q/K/V are concatenated.
2. **Compiling arithmetic into gated FF**: The hard part. Must encode nonlinear operations (ADD, MUL, comparisons) into `relu(gate) * val` form. The gate selects which opcode is active; the val routes the correct computation.
3. **Address verification**: Currently uses dedicated "address checking" heads. In MHA format, these become additional head slots in the same layer.

## Key Files

- `executor.py` — add `VanillaTransformer` class (keep existing `CompiledModel` for reference)
- `isa.py` — token vocab (from dependency)

## Dependencies

- Token vocabulary issue (provides `nn.Embedding` + vocab)

## Complexity

**Large** — This is the hardest issue in the series. Requires mapping custom attention heads into standard MHA format and compiling opcode dispatch + arithmetic into gated FF weight matrices.
